src/NUnitFramework/framework/Internal/TestParameters.cs

```diff
@@ -116,30 +116,42 @@ private void InitializeArguments(object[] args)
         /// </summary>
         public string TestName { get; set; }
 
+        /// <summary>
+        /// If not <see langword="null"/>, overrides the list of argument display names to be used when generating a test name.
+        /// </summary>
+        // TODO: use private protected as soon as the codebase supports C# 7.2
+        internal string[] ArgDisplayNames { private get; set; }
+
         /// <summary>
         /// Gets the property dictionary for this test
         /// </summary>
         public IPropertyBag Properties { get; private set; }
 
         #endregion
 
-        #region IApplyToTest Members
-
         /// <summary>
-        /// Applies ParameterSet values to the test itself.
+        /// Applies the encapsulated parameters to the test.
         /// </summary>
-        /// <param name="test">A test.</param>
-        public void ApplyToTest(Test test)
+        public void ApplyToTest(Test test, TestNameGenerator defaultTestNameGenerator)
         {
-            if (this.RunState != RunState.Runnable)
-                test.RunState = this.RunState;
+            if (RunState != RunState.Runnable)
+                test.RunState = RunState;
 
             foreach (string key in Properties.Keys)
                 foreach (object value in Properties[key])
                     test.Properties.Add(key, value);
```

Comment:
I find these two things being done in both TestParameters and TestFixtureBuilder-but-only-under-certain-circumstances kinda confusing. Is there a reason they need to be done at this point? Setting the RunState doesn't feel like anything to do with TestParameters? (I know this isn't your code, but while we're making big changes in this area! 😄)

Reply:
`TestParameters` has a `RunState` because it can include properties like `Ignore` and can end up as `RunState.Invalid` if the parameters don't match the method to which the parameter applies.


Basically there are a number of components, including `TestParameters` and `TestFixtureBuilder`, which can detect problems calling for setting the `RunState`. Various attributes do the same thing.

Reply:
Ah, I hadn't realised TestParameters also kept a RunState.

Reply:
> TestParameters and TestFixtureBuilder-but-only-under-certain-circumstances

I am sympathetic to this and would have made TestFixtureBuilder consistent with TestCaseBuilder already (always having a TestParameters instance no matter what) but I was afraid that didn't belong in this PR. See https://github.com/nunit/nunit/pull/2538#discussion_r156254117.